### ISSUE TITLE
Add icon + accessible tooltip to links that open in new window

### ISSUE THREAD

**Issue by enriquesanchez** [WordPress/gutenberg#20194]
Branching off from [this suggestion made](https://github.com/WordPress/gutenberg/pull/18742#issuecomment-565241980) in #18742.

Links that open in a new tab or window need to be communicated as such to all users in an accessible way.

In particular, we want the alert to be perceived by all users (not just screen reader users) and such to be non-obtrusive to users who are sensitive to content overload and repetition.

It was suggested to [use the already common practice of adding an icon next to the link](https://github.com/WordPress/gutenberg/pull/18742#issuecomment-565663248) and in addition to follow this [accessible example from Inclusive Components](https://inclusive-components.design/tooltips-toggletips/#tooltipasauxiliarydescription) for accessible tooltips. We think this approach will get us what we need.

**Comment by enriquesanchez** [comment]
cc: @ellatrix and @MarcoZehe because I know we were all discussing this in #18742. Feel free to add any clarification or additional comments or questions.

**Comment by enriquesanchez** [comment]
I will be adding design mockups shortly.

**Comment by enriquesanchez** [comment]
Here are a couple of mockups that hopefully clarify the intent. Any link that is configured to open on a new tab or window, should have an icon next to it that indicates this:

<img width="773" alt="Screen Shot 2020-02-12 at 12 02 44" src="https://user-images.githubusercontent.com/3276087/74364263-d6f1f800-4d91-11ea-8375-427485c72c50.png">

(Alt: the image above shows a text paragraph with a link in it. The link has the "opens in a new window" icon next to it.)

On hover, the link should display a tooltip that textually indicates the action: "Opens in a new window".

<img width="814" alt="Screen Shot 2020-02-12 at 12 17 24" src="https://user-images.githubusercontent.com/3276087/74364599-11f42b80-4d92-11ea-951b-aa5d40a2e601.png">

(Alt: the image above shows a text paragraph with a link being hovered. There is a tooltip floating above the link with the label "Opens in a new window".)

**Comment by paaljoachim** [comment]
Hi Enrique @enriquesanchez 

Personally I find it very visually distractive having an icon beside a link for instance in the middle of the text. Or for instance having a list of links with icons beside each link that all open in new tabs.

I think it is time for an Accessibility mode option in Gutenberg. Turning on the Accessibility option would turn the icon beside each link on along with various other helpful features. This way the user can choose to have accessibility mode on or off.

**Comment by pwkip** [comment]
I completely agree that accessibility is essential. But in my opinion, adding visual elements to the front-end should never be forced by WordPress. It should be the responsibility of the end user or the theme (but I would still not recommend themes to adopt this by default). I also figure that people who value these kind of tools, would want to have them for every website, and not only for WordPress websites, so these people will probably already have a browser extension installed that adds the icon to all links automatically, for example: 
- https://chrome.google.com/webstore/detail/external-followed-link-hi/ldppbkmjichlgoknicpibbekkajojclb?hl=en-GB
- https://chrome.google.com/webstore/detail/targetblank-toggler/emjjmkkcdllendgbldliiphlbpnomnjk

Maybe not the best analogy, so apologies in advance, but we should also not have WordPress automatically start narrating the website by default because some users are visually impaired. These users would install some external screen reader software. I think the same holds true in this case. People who value this feature should enable it trough their browser. (just my thought on this. I'm by no means an accessibility expert)

Lastly, from a technical point of view, it is very easy for themes to add icons to links with just a couple of lines of CSS. So it's a feature that can easily be enabled by themes that value this feature. This also means that it can easily be disabled. So even if WordPress would force this feature, theme authors would still want control over it, and will likely disable if they think it does not fit with their theme.

**Comment by enriquesanchez** [comment]
@paaljoachim @pwkip I disagree here. 

When links are opened in a new window, is is important that there is advance warning and proper communication of context change. Failing to do so can confuse all kinds of users.

From [wordpress.org/about/accessibility/](https://wordpress.org/about/accessibility/):

>WordPress aims to make the WordPress Admin and bundled themes fully WCAG 2.0 AA compliant where possible.

This is a WCAG requirement:

- [Opening new windows and tabs from a link only when necessary](https://www.w3.org/TR/WCAG20-TECHS/G200.html)
- [Using the target attribute to open a new window on user request and indicating this in link text](https://www.w3.org/TR/WCAG20-TECHS/H83.html)

----------

>I think it is time for an extra accessibility/extra visual mode option in Gutenberg

>People who value this feature should enable it through their browser

I think the opposite is better. Good accessibility is good usability and it should come by default. It's important to understand that accessibility benefits everyone, not just those we traditionally identify as disabled users.

**Comment by pwkip** [comment]
> WordPress aims to make the **WordPress Admin** and **bundled themes** fully WCAG 2.0 AA compliant where possible.

Ooh.. I thought this style was going to be forced on **all** front-end themes. That would be a bad idea in my opinion - and it actually wouldn't even be possible given the architecture of WP now I come to think of it. But for WP twenty-something themes and the WP backend, I really don't mind. Some users, that have been running a twenty-ten theme for over 10 years, will likely find it a bit annoying that all of a sudden all their external links get an icon added though, so maybe this should only be considered for upcoming themes?

**Comment by carolinan** [comment]
I have reservations about the tooltip because it does cover other text, but title attributes are not accessible.

**Comment by aristath** [comment]
I think there is a misunderstanding.... 
These icons do **not** indicate that a link opens in a new tab.
The icon indicates an *external* link, so one that is not on the same site/domain.

These is a separate icon for opens-in-new-window behaviour, and you can see their difference here: https://designnotes.blog.gov.uk/2016/11/28/removing-the-external-link-icon-from-gov-uk/ as well as some extra info.

**Comment by enriquesanchez** [comment]
Thanks for that clarification @aristath

**Comment by LukaszJaro** [comment]
Equalize digital has a plugin they built more recently that has a tooltip, icon and text warning https://wordpress.org/plugins/accessibility-new-window-warnings/

![image](https://user-images.githubusercontent.com/10998875/138899576-a44798c3-dd50-4ea5-85e3-23a2aabf62e8.png)

![image](https://user-images.githubusercontent.com/10998875/138899847-df80955b-ac79-4169-89d7-7e07fe79ccec.png)

**Comment by amberhinds** [comment]
Just seeing this issue.  I would love to see these warnings included in core automatically.

**Comment by marcusquinn** [comment]
@enriquesanchez mock-up looks great, and I see no reason why, being a unicode symbol, this couldn't be a core accessibility option for WP in the Customiser.

The plugin @LukaszJaro kindle references is a slightly less polished implementation, but overreaching in that it applies itself to icon buttons, and the tooltips don't consider the edges of the screen.

Making this a WP core feature would solve the issue in a more standard, universal way, to save every theme from re-inventing this need.

**Comment by SteelWagstaff** [comment]
I agree with @marcusquinn and appreciate the work @enriquesanchez. Are there any plans to include this in a future release of WP core? If not, does anyone have any suggestions for how to implement this as the default behaviour for WordPress sites which would like to programmatically announce when links open in a new window/tab that don't involve creating a child theme or purchasing a premium plugin?

**Comment by amberhinds** [comment]
@SteelWagstaff in the free version of our Accessibility Checker Plugin, there is a setting that you can enable to turn this on automatically. You [download the plugin from WordPress.org](https://wordpress.org/plugins/accessibility-checker/), and this feature doesn't require a premium plugin purchase.

**Comment by SteelWagstaff** [comment]
Thanks @amberhinds -- love the work you're doing at Equalize Digital!
